pkg/reconciler/instances/ory/db/db.go

```diff
@@ -60,9 +61,27 @@ func (c *Config) preparePostgresDSN() string {
 	return fmt.Sprintf(dsnTemplate, "postgres", c.Global.PostgresCfg.User, c.Global.PostgresCfg.Password, postgresURL, c.Global.PostgresCfg.DBName, dsnOpts)
 }
 
+func (c *Config) prepareMySQLDSN() string {
+	dbURL := fmt.Sprintf("tcp(%s)", c.Global.Ory.Hydra.Persistence.URL)
+	return fmt.Sprintf(dsnTemplate, c.Global.Ory.Hydra.Persistence.DBType, c.Global.Ory.Hydra.Persistence.Username, c.Global.Ory.Hydra.Persistence.Password, dbURL, c.Global.Ory.Hydra.Persistence.DBName, mysqlDsnOpts)
+}
+
+func (c *Config) prepareGenericDSN() string {
+	return fmt.Sprintf(dsnTemplate, c.Global.Ory.Hydra.Persistence.DBType, c.Global.Ory.Hydra.Persistence.Username, c.Global.Ory.Hydra.Persistence.Password, c.Global.Ory.Hydra.Persistence.URL, c.Global.Ory.Hydra.Persistence.DBName, dsnOpts)
+}
+
 func (c *Config) prepareStringData() map[string]string {
 	if c.Global.Ory.Hydra.Persistence.Enabled {
-		return map[string]string{"secretsSystem": generateRandomString(32), "secretsCookie": generateRandomString(32), "dsn": c.preparePostgresDSN(), "postgresql-password": c.Global.PostgresCfg.Password, "postgresql-replication-password": generateRandomString(10)}
+		if c.Global.Ory.Hydra.Persistence.PostgresqlFlag.Enabled {
+			return map[string]string{"secretsSystem": generateRandomString(32), "secretsCookie": generateRandomString(32), "dsn": c.preparePostgresDSN(), "postgresql-password": c.Global.PostgresCfg.Password, "postgresql-replication-password": generateRandomString(10)}
```

Comment:
Is there any reason for this password having 10 signs? Maybe it should have more?